### PR TITLE
Some corrections on principal CMakeLists to use a dynawo distrib.

### PR DESCRIPTION
**Pull request by gautierbureau**

### CODE REVIEW

**Review by rosiereflo — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -77,25 +77,31 @@ set_optimization()
...
 list(APPEND CMAKE_PREFIX_PATH "${DYNAWO_HOME}/share/cmake")
 list(APPEND CMAKE_MODULE_PATH "${DYNAWO_HOME}/cmake")
 list(APPEND CMAKE_MODULE_PATH "${DYNAWO_HOME}/share/cmake")
 
-#LibArchive
-find_package(LibArchive 2.8.0 REQUIRED)
-#ZLIB
+# ZLIB
 find_package(ZLIB 1.2.3 REQUIRED)
+set_property(TARGET ZLIB::ZLIB APPEND PROPERTY
```
Comment (lecourtoisflo) [inline CMakeLists.txt:90]: I don't get the point of this line. Does find_package already say that ZLIB is found in ZLIB_LIBRARY

Comment (gautierbureau) [reply CMakeLists.txt:90]: The target in not well formed and we need imported location to be valid to link with a static distribution of dynawo (same is done in dynawo)

File: CMakeLists.txt [hunk 2]
```diff
@@ -135,7 +141,7 @@ else()
 endif()
 # Tell BOOST to disable auto linking
 add_definitions(-DBOOST_ALL_NO_LIB)
-find_package(Boost 1.64.0 REQUIRED COMPONENTS program_options filesystem system)
+find_package(Boost 1.64.0 REQUIRED COMPONENTS program_options filesystem system log serialization)
```
[the remark this answers is not in the document]
Comment (lecourtoisflo) [reply CMakeLists.txt:144]: same question for serialization

[the remark this answers is not in the document]
Comment (gautierbureau) [reply CMakeLists.txt:144]: with a static dynawo distrib we get the error 
```
CMake Error at sources/CMakeLists.txt:26 (add_executable):
  Target "DynaFlowLauncher" links to target "Boost::log" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at sources/CMakeLists.txt:26 (add_executable):
  Target "DynaFlowLauncher" links to target "Boost::serialization" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?
```

**Review by lecourtoisflo — APPROVED** [review]

**Pull request merged**